### PR TITLE
`Assessment`: Allow instructors to select csv export options before downloading course scores

### PR DESCRIPTION
**Pull request by maximiliansoelch**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).

#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-tests/).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context

When course scores get downloaded, the field separator as well as the quote strings are currently hardcoded and can not be customized by the user.

### Description

When the user clicks on the export button for downloading course scores, a new modal view is opened through which a custom field separator as well as custom quote strings can be selected.

Currently the newly added components are only used when exporting course scores, but in general they can be used everywhere in the project where csv exports are possible.

### Steps for Testing

Prerequisites:
- 1 Course
- 1 Student
- 1 Exercise, on which the student participated

1. Log in to Artemis
2. Navigate to Course Administration, select the course, then Scores 
3. Click the `Export as CSV` button
4. Select a field separator as well as a the quote string that should be used for generating the csv file
5. Click `Export` 
6. View the downloaded csv file in a text editor and make sure the previously selected options were applied 

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage

| Class/File | Branch | Line |
|------------|-------:|-----:|
| csv-export-button.component.ts | 75% | 100% |
| csv-export-modal.component.ts | 85.71% | 96.42% |
| course-scores.component.ts | 73.84% | 94.72% |

### Screenshots

csv-export

### CODE REVIEW

**Review by b-fein — COMMENTED** [review]
Works as expected when testing with various export options. Nice work, especially already preparing that the same settings can be easily re-used in the future for the other exports. :+1:

I left two suggestions in the code. I did not do a detailed code-review of the HTML changes, though, as others are more familiar with best practices for the layout.

+1 on the quoting of the column headers as already suggested. As this is handled by the CSV library, though, I’m not sure if that can be easily fixed.
⇒ If it can be implemented, then I’d like you to. If that is a limitation by the library, then I’d approve without changing that, as the remaining functionality works as intended.

File: src/main/webapp/app/shared/export/csv-export-modal.component.html [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+                    >
+                        {{ 'export.csvFile.quoteStrings.none' | artemisTranslate }}
+                    </div>
+                </div>
+            </div>
+        </div>
+    </div>
+    <div class="modal-footer justify-content-between">
+        <div class="flex-grow-1 d-flex justify-content-end">
+            <button type="button" class="btn btn-default cancel" data-dismiss="modal" (click)="cancel()">
```
[the remark this answers is not in the document]
Comment (maximiliansoelch) [reply src/main/webapp/app/shared/export/csv-export-modal.component.html]: Adapted

File: src/main/webapp/app/shared/export/csv-export-modal.component.html [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+            <label class="form-control-label" jhiTranslate="export.csvFile.fieldSeparator.label">Field Separator</label>
+            <div>
+                <div class="btn-group">
+                    <div
+                        class="btn"
+                        [ngClass]="{ 'btn-primary': options.fieldSeparator === CsvFieldSeparator.COMMA, 'btn-default': options.fieldSeparator !== CsvFieldSeparator.COMMA }"
+                        (click)="setCsvFieldSeparator(CsvFieldSeparator.COMMA)"
+                    >
+                        {{ 'export.csvFile.fieldSeparator.comma' | artemisTranslate }}
+                    </div>
```
[the remark this answers is not in the document]
Comment (maximiliansoelch) [reply src/main/webapp/app/shared/export/csv-export-modal.component.html]: Code changed to your suggestion.

**Comment by maximiliansoelch** [comment]
@bassner Thanks for the comment. I actually thought the export library will take care of that and therefore missed that the header row is not quoted according to the user's specified setting. I adjusted it now so that the user requested quotestring is also applied to the header row.

@b-fein Thanks for the review and testing. I implemented the changes according to your suggestions regarding the space between the export and cancel button as well as now utilizing ngfor to generate the button groups.

**Review by valeriebucher — APPROVED** [review]
Export works as expected!

**Review by bassner — APPROVED** [review]
Thanks for adding the changes!
The CSV column titles are now properly quoted.
Selected options are applied correctly, and the export works correctly.

Code looks good to me as well 👍

**Pull request merged**